client/blocks/simple-site-rename-form/index.jsx

```diff
@@ -0,0 +1,91 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import { noop, get, flow } from 'lodash';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import CompactCard from 'components/card/compact';
+import FormButton from 'components/forms/form-button';
+import FormTextInputWithAffixes from 'components/forms/form-text-input-with-affixes';
+import ConfirmationDialog from './dialog';
+
+export class SimpleSiteRenameForm extends Component {
+	state = {
+		showDialog: false,
+		domainFieldValue: '',
+	};
+
+	onSubmit = event => {
+		event.preventDefault();
+
+		this.setState( {
+			showDialog: true,
+		} );
+	};
+
+	onDialogClose = () => {
+		this.setState( {
+			showDialog: false,
+		} );
+	};
+
+	onFieldChange = event => {
+		this.setState( { domainFieldValue: get( event, 'target.value' ) } );
+	};
+
+	render() {
+		const { currentDomainName, translate } = this.props;
+
+		// This may need more consideration given the new x.music.blog type domains
+		const isWPCOM = currentDomainName.type === 'WPCOM';
```

Comment:
This check (& other places `currentDomainName` is accessed) needs to be guarded per `currentDomainName` being set.

I'm seeing `Uncaught TypeError: Cannot read property 'type' of undefined` when browsing directly to https://calypso.live/devdocs/blocks?branch=add%2Fsimple-sites-rename-form

It's likely why the e2e test is failing as well.